mods/stairs/init.lua

```diff
@@ -256,355 +373,425 @@ if replace then
 end
 
 
--- Stair/slab registration function.
--- Nodes will be called stairs:{stair,slab}_<subname>
+-- Stair/slab/stairi/stairo registration function.
+-- Nodes will be called stairs:{stair,slab,stairi,stairo}_<subname>
 
-function stairs.register_stair_and_slab(subname, recipeitem,
-		groups, images, desc_stair, desc_slab, sounds)
+function stairs.register_stair_and_slab_and_stairi_and_stairo(subname, recipeitem,
+		groups, images, desc_stair, desc_slab, desc_stairi, desc_stairo, sounds)
 	stairs.register_stair(subname, recipeitem, groups, images, desc_stair, sounds)
 	stairs.register_slab(subname, recipeitem, groups, images, desc_slab, sounds)
+    stairs.register_stairi(subname, recipeitem, groups, images, desc_stairi, sounds)
+	stairs.register_stairo(subname, recipeitem, groups, images, desc_stairo, sounds)
 end
 
 
--- Register default stairs and slabs
+-- Register default stairs and slabs and corners
 
-stairs.register_stair_and_slab(
+stairs.register_stair_and_slab_and_stairi_and_stairo(
```

Comment:
I saw MarkuBu's reply, so I will keep pushing on with this as best I can.
I have addressed the issues, aside from the registration function, with backward compat.
Unfortunately, after trying several methods (including the one you suggest paramat), none work.
The only code I have seen which works perfectly, is the registration function from tenplus1's stairs redo. (which also changes the way each material is accomplished)
Trying the mentioned method here, only 3 nodes out of 4 appear (and this varies, related to which is in the code first)
In one case, there are no slabs, yet the inner corner is labeled as "slab", etc.. And this changes depending on how it is coded. And I have (IMO) exhausted all methods, with these suggestions in mind.